crates/bevy_text/src/text2d.rs

```diff
@@ -226,3 +228,148 @@ pub fn update_text2d_layout(
 pub fn scale_value(value: f32, factor: f32) -> f32 {
     value * factor
 }
+
+/// System calculating and inserting an [`Aabb`] component to entities with some
+/// [`TextLayoutInfo`] and [`Anchor`] components, and without a [`NoFrustumCulling`] component.
+///
+/// Used in system set [`VisibilitySystems::CalculateBounds`](bevy_render::view::VisibilitySystems::CalculateBounds).
+pub fn calculate_bounds_text2d(
+    mut commands: Commands,
+    text_to_update_aabb: Query<
+        (Entity, &TextLayoutInfo, &Anchor),
+        (
+            Or<(Without<Aabb>, Changed<TextLayoutInfo>)>,
+            Without<NoFrustumCulling>,
+        ),
+    >,
+) {
+    for (entity, layout_info, anchor) in &text_to_update_aabb {
+        // `Anchor::as_vec` gives us an offset relative to the text2d bounds, by negating it and scaling
+        // by the logical size we compensate the transform offset in local space to get the center.
+        let center = (-anchor.as_vec() * layout_info.logical_size)
+            .extend(0.0)
+            .into();
+        // Distance in local space from the center to the x and y limits of the text2d bounds.
+        let half_extents = (layout_info.logical_size / 2.0).extend(0.0).into();
+        commands.entity(entity).try_insert(Aabb {
```

Comment:
Could we try to directly mutate `Aabb` here where available? `Option<&mut Aabb>` in the query, and adding a command if and only if it's not present can avoid potentially adding a large number of commands.